flask_app/tasks.py

```diff
@@ -111,30 +113,35 @@ def create_key(s: str) -> str:
 Montgomery Scott
 """
 
-def _get_combadge_by_version(combadge_version: str) -> bytes:
-    return {
-        "v1": _COMBADGE_V1,
-        "v2": _COMBADGE_V2,
-    }[combadge_version]
-
-
-def _upload_combadge(ssh_client: SSHClient, combadge_version: str) -> str:
-    _, stdout, stderr = ssh_client.exec_command("mktemp /tmp/combadge.XXX")
+def _get_os_type(ssh_client: SSHClient) -> str:
+    _, stdout, stderr = ssh_client.exec_command("uname")
     retcode = stdout.channel.recv_exit_status()
     if retcode != 0:
-        raise Exception(stderr.read().decode("utf-8"))
+        return 'windows'
+    return stdout.read().decode("utf-8").strip().lower()
 
-    combadge_path = stdout.read().decode("utf-8").strip()
 
-    stdin, stdout, stderr = ssh_client.exec_command(f"sh -c \"cat > {combadge_path} && chmod +x {combadge_path}\"")
-    combadge = _get_combadge_by_version(combadge_version)
-    stdin.write(combadge)
-    stdin.channel.shutdown_write()
-    retcode = stdout.channel.recv_exit_status()
-    if retcode != 0:
-        raise Exception(stderr.read().decode("utf-8"))
+def _generate_random_combadge_name(stringLength: int) -> str:
+    random_string = ''.join(random.choice(string.ascii_lowercase) for i in range(stringLength))
+    return f"combadge_{random_string}"
+
+
+def _upload_combadge(ssh_client: SSHClient, combadge_version: str) -> str:
+    os_type = _get_os_type(ssh_client)
+    combadge_name = _generate_random_combadge_name(stringLength=10)
+    combadge_type_identifier = combadge_version if combadge_version == 'v1' else os_type
+    is_nt = combadge_type_identifier == 'windows'
```

Comment:
I'd suggest renaming to `is_windows`, as that's exactly what you check (`.._identifier == ...`)